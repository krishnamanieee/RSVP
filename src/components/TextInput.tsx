import React from 'react';
import {
  TextInputProps,
  TextInput as RNTextInput,
  View,
  Text,
  StyleSheet,
} from 'react-native';
import Theme from '../theme';

export interface ITextInputProps extends TextInputProps {
  errorMessage?: string;
  title?: string;
}

class TextInput extends React.PureComponent<ITextInputProps, any> {
  render() {
    const {errorMessage, title, style, ...props} = this.props;
    return (
      <View style={styles.container}>
        {title && title.length > 0 && <Text style={styles.title}>{title}</Text>}
        <RNTextInput style={[styles.textInput, style]} {...props} />
        {errorMessage && errorMessage.length > 1 ? (
          <Text style={styles.errorText}>{errorMessage}</Text>
        ) : null}
      </View>
    );
  }
}

export default TextInput;

const styles = StyleSheet.create({
  container: {
    marginVertical: 5,
    marginHorizontal: 5,
  },
  textInput: {
    backgroundColor: Theme.colors.input.background,
    paddingVertical: 10,
    paddingHorizontal: 5,
  },
  title: {
    paddingVertical: 5,
    fontSize: Theme.font.text.text,
    color: Theme.colors.input.TitleText,
  },
  errorText: {
    paddingVertical: 5,
    fontSize: Theme.font.text.label,
    color: Theme.colors.input.errorText,
  },
});
