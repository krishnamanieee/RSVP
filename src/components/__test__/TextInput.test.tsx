import React from 'react';
import renderer from 'react-test-renderer';
import TextInput, {ITextInputProps} from '../TextInput';
let props: ITextInputProps = {
  errorMessage: '',
};
describe('TextInput', () => {
  it('should match match snapshot', () => {
    const tree = renderer.create(<TextInput {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
