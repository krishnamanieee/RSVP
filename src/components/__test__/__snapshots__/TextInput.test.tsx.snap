// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match match snapshot 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 5,
      "marginVertical": 5,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "backgroundColor": "#dadada",
          "paddingHorizontal": 5,
          "paddingVertical": 10,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`TextInput should match match snapshot 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 5,
      "marginVertical": 5,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "backgroundColor": "#dadada",
          "paddingHorizontal": 5,
          "paddingVertical": 10,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
